### ISSUE TITLE
Support of fermionic systems

### ISSUE THREAD

**Issue by diret47** [QuantumKitHub/TensorKit.jl#96]
Hi, Jutho.
Is it possible to compute fermionic PEPS with `TensorKit`? For example Hubbard model?

**Comment by Jutho** [comment]
Hi @diret47 . TensorKit.jl supports tensors with fermionic symmetries, such that e.g. the necessary signs are automatically included upon permuting tensor legs.

However, TensorKit is a low-level library and does not contain higher level tensor network algorithms. For this, you should look at for example MPSKit.jl and PEPSKit.jl. Support for fermions in PEPSKit.jl is not yet included, but something we will be actively working on in the near future.

**Comment by diret47** [comment]
> @diret47, I've recently implemented the contraction of fermionic PEPS in [this](https://github.com/quantumghent/PEPSKit.jl/tree/planar_branch) branch of PEPSKit.jl .
> 
> There is currently no support for optimization of states as this would require evaluation of the gradient which is a formidable task... The plan is to bypass this trough automatic differentiation but this first requires some updates within TensorKit.jl and TensorOperations.jl .

That's awesome. I suppose we can already contract fPEPS by PEPSKit.jl easily. I hope implementation of AD optimization would be updated soon. I think one can use some SVD fashion optimization as a expedient now.

**Comment by Gertian** [comment]
@diret47 , what do you mean with SVD fashion optimization ?

**Comment by diret47** [comment]
> @diret47 , what do you mean with SVD fashion optimization ?

Sorry, I just mean that sth like full update and simple update in iPEPS. I suppose that the optimization like DMRG may also be used in iPEPS, but it's not exact.

**Comment by Confusio** [comment]
Is there any document on TensorKit that explains the details of treating the fermion signs and swap operations in PEPS?

**Comment by Gertian** [comment]
@lkdvos , @diret47 this is now implemented in PEPSKit right ?

**Comment by diret47** [comment]
> @lkdvos , @diret47 this is now implemented in PEPSKit right ?

Really I found that there is an new test /test/pwave.jl on p-wave superconductivity. Seems Hubbard model is not implemented now but I think it might come soon.

**Comment by lkdvos** [comment]
In principle this should indeed all just work, I quickly added the Hamiltonian in a draft PR for convenience, but keep in mind that I haven't checked this. If you could compare this to some cases you know the answers for, I would be very happy to hear the results.

https://github.com/QuantumKitHub/PEPSKit.jl/pull/80

**Comment by WiseLu** [comment]
@Jutho  Hello Jutho,
Does the "ncon" in TensorOperation.jl support the fermionic contraction?

**Comment by Jutho** [comment]
Yes, it should be equivalent to the corresponding `@tensor` call. We hope to update the TensorKit documentation soon to provide more information about the fermionic (and anyonic) tensor contractions
